consensus/polybft/signer/signature_test.go

```diff
@@ -26,6 +27,54 @@ func Test_VerifySignature(t *testing.T) {
 	assert.False(t, signature.Verify(blsKey.PublicKey(), invalidTestMsg))
 }
 
+func Test_VerifySignature_NegativeCases(t *testing.T) {
+	t.Parallel()
+
+	validTestMsg := testGenRandomBytes(t, 32)
+
+	blsKey, _ := GenerateBlsKey()
+	signature, err := blsKey.Sign(validTestMsg)
+	require.NoError(t, err)
+
+	require.True(t, signature.Verify(blsKey.PublicKey(), validTestMsg))
+
+	t.Run("Wrong public key", func(t *testing.T) {
+		t.Parallel()
+
+		publicKey := blsKey.PublicKey()
+		publicKey.g2.Add(publicKey.g2, publicKey.g2)
+
+		require.False(t, signature.Verify(publicKey, validTestMsg))
+	})
+
+	t.Run("Tampered message", func(t *testing.T) {
+		t.Parallel()
+
+		msgCopy := make([]byte, len(validTestMsg))
+		copy(msgCopy, validTestMsg)
+
+		for i := 0; i < len(msgCopy); i++ {
+			b := msgCopy[i]
+			msgCopy[i] = b + 1
+
+			require.False(t, signature.Verify(blsKey.PublicKey(), msgCopy))
+			msgCopy[i] = b
+		}
+	})
+
+	t.Run("Tampered signature", func(t *testing.T) {
+		t.Parallel()
+
+		sigCopy := signature
+		sigCopy.g1.Add(sigCopy.g1, sigCopy.g1) // change signature
```

Comment:
Maybe include random generation of G1 as well if that makes sense (not sure if only that test case can substitute all the others). Also if you can think of any other adversarial manipulation with signatures, I'd consider adding it as well, so that no matter how we change signature, it is going to be invalid.

```suggestion
		sigCopy := signature
		sigCopy.g1.Add(sigCopy.g1, sigCopy.g1) // change signature
		require.False(t, sigCopy.Verify(blsKey.PublicKey(), validTestMsg))

		sigCopy = signature
		sigCopy.g1.ScalarMult(sigCopy.g1, big.NewInt(2)) // change signature
		require.False(t, sigCopy.Verify(blsKey.PublicKey(), validTestMsg))

        _, g1, err := bn256.RandomG1(rand.Reader)
		require.NoError(t, err)
		sigCopy.g1 = g1
		require.False(t, sigCopy.Verify(blsKey.PublicKey(), validTestMsg))
```

Reply:
https://github.com/0xPolygon/polygon-edge/pull/1214/commits/19a511499342a773c277798166b2cc5feef543ec